types/src/waypoint.rs

```diff
@@ -144,6 +144,45 @@ impl Ledger2WaypointConverter {
     }
 }
 
+impl<'de> Deserialize<'de> for Waypoint {
+    fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        if deserializer.is_human_readable() {
+            let s = <String>::deserialize(deserializer)?;
+            Waypoint::from_str(&s).map_err(D::Error::custom)
+        } else {
+            // In order to preserve the Serde data model and help analysis tools,
+            // make sure to wrap our value in a container with the same name
+            // as the original type.
```

Comment:
I'm not sure I have enough context to fully understand this, but, if we're changing the WaypointConfig enum, why not also update any "analysis tools" or dependencies, so we don't need to support this edge case?

Reply:
so the question is what's the right representation in a binary format... I don't know, we don't dictate that nor share it anywhere, so I suspect just being explicit about the format and leveraging the underlying style should be good, but you know, I don't want to rock that boat.